2008-04-14  David S. Miller  <davem@davemloft.net>

 * sysdeps/unix/sysv/linux/sparc/sparc32/sem_wait.c
 (__old_sem_wait): Fix argument to lll_futex_wait().

===================================================================
RCS file: /cvs/glibc/libc/nptl/sysdeps/unix/sysv/linux/sparc/sparc32/sem_wait.c,v
retrieving revision 1.2
retrieving revision 1.3
diff -u -r1.2 -r1.3
--- libc/nptl/sysdeps/unix/sysv/linux/sparc/sparc32/sem_wait.c	2007/08/15 20:47:42	1.2
+++ libc/nptl/sysdeps/unix/sysv/linux/sparc/sparc32/sem_wait.c	2008/04/15 15:00:30	1.3
@@ -155,7 +155,7 @@
       /* Enable asynchronous cancellation.  Required by the standard.  */
       int oldtype = __pthread_enable_asynccancel ();
 
-      err = lll_futex_wait (futex, 0,
+      err = lll_futex_wait (&isem->value, 0,
 			    isem->private ^ FUTEX_PRIVATE_FLAG);
 
       /* Disable asynchronous cancellation.  */
